src/main/java/loci/common/StreamHandle.java

```diff
@@ -49,6 +52,19 @@
  */
 public abstract class StreamHandle implements IRandomAccess {
 
+  // TODO: Decide how to handle S3Handle and other reader settings
+  public static class Settings {
```

Comment:
It seems a little odd to have this in ```StreamHandle``` when there is no other API.  Maybe having ``` public static String cacheObject(String url, Settings s) throws IOException, HandleException``` be abstract is enough, but if there are no real plans to support this for non-S3 paths then it might be clearer to move to ```S3Handle```.

Reply:
My thinking was you'd want the option of caching for any remote protocols, not just S3.